### ISSUE TITLE
DNS record change is not reflected in Cloudflare

### ISSUE THREAD

**Issue by makered** [hashicorp/terraform#7690]
Terraform: v0.6.16
Resource: cloudflare_record

I was able to successfully create a CNAME record on Cloudflare. Later I needed to change the DNS name so I modified the `name` attribute and applied the change. Result was:

```
module.elb.cloudflare_record.cf_dns_record_elb: Modifying...
  name: "oldname" => "newname"
module.elb.cloudflare_record.cf_dns_record_elb: Modifications complete
```

However, the Cloudflare record was unaltered and the output showed:

```
Outputs:
  cloudflare_elb_fqdn       = oldname.mydomain.com
```

HCL:

```
variable "dns_hostname" {
  description = "ELB DNS hostname"
  default = "newname"
}
```

```
resource "cloudflare_record" "cf_dns_record_elb" {
  domain = "${var.cloudflare_domain}"
  name = "${var.dns_hostname}"
  value = "${aws_elb.elb.dns_name}"
  type = "CNAME"
}
output "cloudflare_elb_fqdn" {
  value = "${cloudflare_record.cf_dns_record_elb.hostname}"
}
```

**Comment by catufunwa** [comment]
I'm still running into this bug with the cloudflare provider on v0.7.3. Is it me or has the cloudflare provider been really flaky the last few version?

**Comment by jlgeering** [comment]
issue still present in Terraform v0.7.13

**Comment by joshuajabbour** [comment]
I can confirm this is present in `0.7.13`. The only workaround I have found is to comment out the individual `cloudflare_record` block so it's destroyed and then uncomment it so it's created anew with the proper name. Unfortunately, there's no `lifecycle` option to force recreation in this manner.

**Comment by mwarkentin** [comment]
Also ran into this on `0.7.13` - wonder if `taint` would work to force a destroy + recreate?

**Comment by mwarkentin** [comment]
Opened https://github.com/cloudflare/cloudflare-go/issues/97 to try and figure out if this is an issue with the upstream library.

**Comment by mwarkentin** [comment]
It looks like this may just be an issue with the upstream Cloudflare library: https://github.com/cloudflare/cloudflare-go/issues/97#issuecomment-268362646

Trying to work with their engineer on a fix here: https://github.com/cloudflare/cloudflare-go/issues/98

**Comment by mwarkentin** [comment]
Tested the fix in https://github.com/cloudflare/cloudflare-go/pull/98, seems to have fixed the issue when using the cloudflare-go library directly - I assume that'll fix the terraform issue as well.

**Comment by mwarkentin** [comment]
@stack72 @catsby The fix for this has been merged upstream - not sure what needs to be done to update the vendored lib. Let me know if there's anything there I can do to help get this closed.

**Comment by stack72** [comment]
Hi @makered - this is closed via #10910

**Comment by makered** [comment]
Perfect, thank you so much for the fix!!!
